src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -388,24 +850,25 @@ class JSONReader
     {
         boost::shared_ptr<const rapidjson::Document> doc;
         // TODO: optimize this. for the first time doc is cached, all the expandable object will never get cached, because we are not update document cache after expand any body
+
         if (_rapidJSONDocuments.find(fullFilename) != _rapidJSONDocuments.end()) {
             doc = _rapidJSONDocuments[fullFilename];
         }
         else {
-            boost::shared_ptr<rapidjson::Document> newDoc;
+            boost::shared_ptr<rapidjson::Document> newDoc = boost::shared_ptr<rapidjson::Document>(new rapidjson::Document());;
```

Comment:
Here you are losing `alloc` again. You don't need these 3 lines of changes. Please revert.